Add line height units example (#30153)
mdn/content: files/en-us/learn/css/building_blocks/values_and_units/index.md

{{EmbedGHLiveSample("css-examples/learn/values-units/em-rem.html", '100%', 1100)}}

#### Line height units

`lh` and `rlh` are relative lenghts units similar to `em` and `rem`. The difference between `lh` and `rlh` is that the first one is relative to the line height of the element itself, while the second one is relative to the line height of the root element, usually `<html>`.

> Using these units we can precisely align box decoration to the text. In the example below we use `lh` unit to create notepad-like lines using [`repeating-linear-gradient()`](/en-US/docs/Web/CSS/gradient/repeating-linear-gradient). It doesn't matter what's the line height of the text, the lines will always start in the right place.

Reviewer: ```suggestion
Using these units, we can precisely align box decoration to the text. In this example, we use `lh` unit to create notepad-like lines using [`repeating-linear-gradient()`](/en-US/docs/Web/CSS/gradient/repeating-linear-gradient). It doesn't matter what's the line height of the text, the lines will always start in the right place.
```

```css hidden
body {
  margin: 0;
  display: grid;
  grid-template-columns: 1fr 1fr;
  padding: 24px;
  gap: 24px;
  background-color: floralwhite;
  font-family: sans-serif;
}

@supports not (height: 1lh) {
  body::before {
    grid-column: 1 / -1;
    padding: 8px;
    border-radius: 4px;
    background-color: tomato;
    color: white;
    content: "You browser doesn’t support lh unit just yet";
  }
}
```

```css
p {
  margin: 0;
  background-image: repeating-linear-gradient(
    to top,
    lightskyblue 0 2px,
    transparent 2px 1lh
  );
}
```

```html
<p style="line-height: 2em">
  Summer is a time for adventure, and this year was no exception. I had many
  exciting experiences, but two of my favorites were my trip to the beach and my
  week at summer camp.
</p>

<p style="line-height: 4em">

>   At the beach, I spent my days swimming, sunbathing, and building sandcastles.

Reviewer: ```suggestion
  At the beach, I spent my days swimming, collecting shells, and building sandcastles.
```
In many cultures, people don't sunbathe. In others, women aren't allowed to. Changing to hopefully be more inclusive.

Reviewer: This example will look better if the text lines are a bit shorter. They're wrapping.

Author: I made the demo a bit taller so it renders without any scrolling starting from a 968px wide screen.

#### Line height units

`lh` and `rlh` are relative lenghts units similar to `em` and `rem`. The difference between `lh` and `rlh` is that the first one is relative to the line height of the element itself, while the second one is relative to the line height of the root element, usually `<html>`.

Using these units, we can precisely align box decoration to the text. In this example, we use `lh` unit to create notepad-like lines using [`repeating-linear-gradient()`](/en-US/docs/Web/CSS/gradient/repeating-linear-gradient). It doesn't matter what's the line height of the text, the lines will always start in the right place.

```css hidden
body {
  margin: 0;
  display: grid;
  grid-template-columns: 1fr 1fr;
  padding: 24px;
  gap: 24px;
  background-color: floralwhite;
  font-family: sans-serif;
}

@supports not (height: 1lh) {
  body::before {
    grid-column: 1 / -1;
    padding: 8px;
    border-radius: 4px;
    background-color: tomato;
    color: white;
    content: "You browser doesn’t support lh unit just yet";
  }
}
```

```css
p {
  margin: 0;
  background-image: repeating-linear-gradient(
    to top,
    lightskyblue 0 2px,
    transparent 2px 1lh
  );
}
```

```html

> <p style="line-height: 2em">
>   Summer is a time for adventure, and this year was no exception. I had many
>   exciting experiences, but two of my favorites were my trip to the beach and my
>   week at summer camp.
> </p>
> 
> <p style="line-height: 4em">
>   At the beach, I spent my days swimming, collecting shells, and building
>   sandcastles. I also went on a boat ride and saw dolphins swimming alongside
>   us.
> </p>

Reviewer: ```suggestion
<p style="line-height: 2em">
  Summer is a time for adventure, and this year was no exception.
  I had many exciting experiences, but two of my favorites were my 
  trip to the beach and my week at summer camp.
</p>

<p style="line-height: 4em">
  At the beach, I spent my days swimming, collecting shells, 
  and building sandcastles. I also went on a boat ride and saw 
  dolphins swimming alongside us.
</p>
```
so the lines don't wrap in the code example
